src/app/cli/src/init/client.ml

```diff
@@ -1789,90 +1789,82 @@ let add_peers_graphql =
                   } ) ) ) )
 
 let compile_time_constants =
-  let genesis_constants = Genesis_constants.Compiled.genesis_constants in
-  let constraint_constants = Genesis_constants.Compiled.constraint_constants in
-  let proof_level = Genesis_constants.Compiled.proof_level in
+  let open Command.Let_syntax in
   Command.async
     ~summary:"Print a JSON map of the compile-time consensus parameters"
-    (Command.Param.return (fun () ->
+    (let%map_open config_file = Cli_lib.Flag.config_files in
+     fun () ->
+       let home = Core.Sys.home_directory () in
+       let conf_dir = home ^/ Cli_lib.Default.conf_dir_name in
+       let genesis_dir =
          let home = Core.Sys.home_directory () in
-         let conf_dir = home ^/ Cli_lib.Default.conf_dir_name in
-         let genesis_dir =
-           let home = Core.Sys.home_directory () in
-           home ^/ Cli_lib.Default.conf_dir_name
-         in
-         let config_file =
-           match Sys.getenv "MINA_CONFIG_FILE" with
-           | Some config_file ->
-               config_file
-           | None ->
-               conf_dir ^/ "daemon.json"
-         in
-         let open Async in
+         home ^/ Cli_lib.Default.conf_dir_name
+       in
+       let open Deferred.Let_syntax in
+       let%map ({ consensus_constants; _ } as precomputed_values), _ =
+         (* This is kind of ugly because we are allowing for supplying a runtime_config value directly, rather than force what is read from the environment *)
+         (* TODO: See if we can initialize consensus_constants without also initializing the ledger *)
          let logger = Logger.create () in
-         let%map ({ consensus_constants; _ } as precomputed_values), _ =
-           let%bind runtime_config =
-             let%map config_file =
-               Runtime_config.Json_loader.load_config_files ~conf_dir ~logger
-                 [ config_file ]
-               >>| Or_error.ok
-             in
-             let default =
-               Runtime_config.of_json_layout
-                 { Runtime_config.Json_layout.default with
-                   ledger =
-                     Some
-                       { Runtime_config.Json_layout.Ledger.default with
-                         accounts = Some []
-                       }
-                 }
-               |> Result.ok_or_failwith
-             in
-             Option.value ~default config_file
-           in
-           Genesis_ledger_helper.Config_loader.init_from_config_file_legacy
-             ~genesis_constants ~constraint_constants ~logger:(Logger.null ())
-             ~proof_level ~cli_proof_level:None ~genesis_dir runtime_config
-           >>| Or_error.ok_exn
+         let%bind m_conf =
```

Comment:
Above logger is initialized as a regular logger, but this is a CLI command, so it's better to set it to `null`, which is done in `compatible`.